sky/packages/sky/lib/rendering/object.dart

```diff
@@ -650,18 +643,20 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
       sky.tracing.end('RenderObject.flushPaint');
     }
   }
-  void initialPaint(ContainerLayer rootLayer, Size size) {
+  void scheduleInitialPaint(ContainerLayer rootLayer) {
     assert(attached);
     assert(parent is! RenderObject);
     assert(!_debugDoingPaint);
     assert(hasLayer);
-    PaintingContext newContext = new PaintingContext.withLayer(rootLayer, Point.origin & size);
-    _paintLayer(newContext);
+    _layer = rootLayer;
+    assert(_needsPaint);
+    _nodesNeedingPaint.add(this);
   }
   void _repaint() {
     assert(hasLayer);
     assert(_layer != null);
-    PaintingContext newContext = new PaintingContext.replacingLayer(_layer, paintBounds);
+    _layer.removeAllChildren();
+    PaintingContext newContext = new PaintingContext.withLayer(_layer, paintBounds);
     _paintLayer(newContext);
```

Comment:
I think you can inline paintLayer() here again now. It was only split out because initialPaint() also used it.